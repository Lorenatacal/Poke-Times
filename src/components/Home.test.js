import React, { Component } from 'react';
import HomeConnected, { Home } from './Home';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
// import configureMockStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() })

test("Home should render correctly with posts", () => {
    const props = {
        posts: [
            {id: '1', title: 'First title', body: 'This is my body'},
            {id: '2', title: 'Second title', body: 'This is the second body'},
            {id: '3', title: 'Third title', body: 'This is my third body'},
        ],
    };
    const wrapper = Enzyme.shallow(<Home posts={props.posts}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
})