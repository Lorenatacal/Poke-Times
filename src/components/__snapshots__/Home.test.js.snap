// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly with posts 1`] = `
<div
  className="container home"
>
  <h4
    className="center"
  >
    Home
  </h4>
  <div
    className="post card"
    key="1"
  >
    <img
      alt="A pokeball"
      src="pokeball.png"
    />
    <div
      className="card-content"
    >
      <Link
        replace={false}
        to="/1"
      >
        <span
          className="card-title red-text"
        >
          First title
        </span>
      </Link>
      <p>
        This is my body
      </p>
    </div>
  </div>
  <div
    className="post card"
    key="2"
  >
    <img
      alt="A pokeball"
      src="pokeball.png"
    />
    <div
      className="card-content"
    >
      <Link
        replace={false}
        to="/2"
      >
        <span
          className="card-title red-text"
        >
          Second title
        </span>
      </Link>
      <p>
        This is the second body
      </p>
    </div>
  </div>
  <div
    className="post card"
    key="3"
  >
    <img
      alt="A pokeball"
      src="pokeball.png"
    />
    <div
      className="card-content"
    >
      <Link
        replace={false}
        to="/3"
      >
        <span
          className="card-title red-text"
        >
          Third title
        </span>
      </Link>
      <p>
        This is my third body
      </p>
    </div>
  </div>
  <AddPost />
</div>
`;

exports[`Home should render correctly without posts 1`] = `
<div
  className="container home"
>
  <h4
    className="center"
  >
    Home
  </h4>
  <div
    className="center"
  >
    No posts yet
  </div>
  <AddPost />
</div>
`;
